R/check_target_dataset_file_ext_unique.R

```diff
@@ -0,0 +1,52 @@
+#' Check that all files of a given target type share a single unique file format
+#'
```

Comment:
Same as in NEWS, I think we can add the detail that only target data are concerned by the test. (It was not obvious to me and I tested with a badly name file with another extension and as expected got no error)

Reply:
And it does not seems to work if you have both partitioned and non-partitioned files. 
In fact, partitioned files do not seem to be identify:
For example, on my test I have:
```r
> hubValidations::check_target_dataset_unique(".")
<message/check_info>
Message:
No time-series target type files detected in target-data directory. Check skipped.
```

and my target-data folder is organized as:
```
|__ target-data
        |_____ target=inc%20death
                              |_______part-0.parquet (same if file called time-series0.parquet)
        |_____ target=inc%20hosp
                              |_______part-0.parquet (same if file called time-series0.parquet)
```

Reply:
@LucieContamin, the reason the partitioned files are not identified is because they need the partitions need to be in a folder called `time-series` (see [2025-01-06 RFC Target Data Formats](https://github.com/reichlab/decisions/blob/main/decisions/2025-01-06-rfc-target-data-formats.md#target-data-location) for a description of target data formats)

```
target-data
|__ time-series
        |_____ target=inc%20death
                              |_______part-0.parquet (same if file called time-series0.parquet)
        |_____ target=inc%20hosp
                              |_______part-0.parquet (same if file called time-series0.parquet)
```

Reply:
It's all good now, so please ignore my second comment, I will just suggests to update the documentation to make it clear what is tested

Reply:
it's ok, I opened an issue in the hubdoc repo, let me know if I can help. I am happy to review or any other help

Reply:
Yeah sorry about that @LucieContamin . The target data work is being rolled out on multiple fronts and the hubDocs documentation has not been updated with respect to some of the later decisions. Definitely an important high impact missing piece though! Perhaps we should discuss at the meeting whether to bump that up the priority scale.